### PR TITLE
[plugins/callback/syslog_json.py] use v2 api, add option to skip sysl…

### PR DESCRIPTION
**Pull request by andrewroffey**
[plugins/callback/syslog_json.py]  use v2 api, add option to skip syslog on gather_facts

##### SUMMARY
syslog_json callback plugin fails during gather_facts on playbooks that run on hosts with many facts, as the syslog message exceeds OS limits. An option should be added to skip syslog on the gather_facts task.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/callback/syslog_json.py

##### ADDITIONAL INFORMATION

Easiest way to reproduce this issue is to run playbook with -vvv and with callback_syslog_json enabled.

```
--- Logging error ---                                                                                                                                                                        
Traceback (most recent call last):                                                                                                                                                           
  File "/usr/lib64/python3.6/logging/handlers.py", line 942, in emit                                                                                                                         
    self.socket.sendto(msg, self.address)                                                                                                                                                    
OSError: [Errno 90] Message too long
<snip>
```

### CODE REVIEW

**Comment by andrewroffey** [comment]
I'm using this patch for my project but if there are other ideas on avoiding the problem I'm open to general feedback.

File: plugins/callback/syslog_json.py [hunk 1]
```diff
@@ -41,6 +41,15 @@
...
             key: syslog_facility
+      setup:
+        description: log setup tasks
+        env:
+          - name: SYSLOG_SETUP
+        type: bool
+        default: true
+        ini:
+          - section: callback_syslog_json
+            key: syslog_setup
```
Comment (felixfontein) [inline plugins/callback/syslog_json.py:52]:
```suggestion
            key: syslog_setup
        version_added: 4.5.0
```

File: plugins/callback/syslog_json.py [hunk 2]
```diff
@@ -41,6 +41,15 @@
         ini:
           - section: callback_syslog_json
             key: syslog_facility
+      setup:
+        description: log setup tasks
```
Comment (felixfontein) [inline plugins/callback/syslog_json.py:45]:
```suggestion
        description: Log setup tasks.
```

File: plugins/callback/syslog_json.py [hunk 3]
```diff
@@ -41,6 +41,15 @@
...
+        env:
+          - name: SYSLOG_SETUP
```
Comment (felixfontein) [inline plugins/callback/syslog_json.py:47]:
```suggestion
          - name: ANSIBLE_SYSLOG_SETUP
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment[(https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)?

I've added a first few (generic) comments below.

**Comment by andrewroffey** [comment]
Thanks for the suggestions! I've merged in those changes and added a changelog fragment.

File: changelogs/fragments/4223-syslog-json-skip-syslog-option.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - syslog_json - add option to skip logging of ``gather_facts`` playbook task; use v2 callback api (https://github.com/ansible-collections/community.general/pull/4223).
```
Comment (felixfontein) [inline changelogs/fragments/4223-syslog-json-skip-syslog-option.yml:2]:
```suggestion
  - syslog_json - add option to skip logging of ``gather_facts`` playbook tasks; use v2 callback API (https://github.com/ansible-collections/community.general/pull/4223).
```

**Comment by felixfontein** [comment]
@imjoseangel do you have comments on this as well?

**Comment by felixfontein** [comment]
I'll merge in a couple of days if nobody complains.

**Comment by felixfontein** [comment]
@andrewroffey thanks for implementing this!

**Pull request merged**